Compile Warning on CentOS 7

Hey, I'm compiling OIIO with gcc 4.8.3 on CentOS 7, and it is failing on the following line:

```
/home/scott/Repositories/oiio/src/libOpenImageIO/imagebufalgo.cpp: In function ‘bool OpenImageIO::v1_6::ImageBufAlgo::convolve(OpenImageIO::v1_6::ImageBuf&, const OpenImageIO::v1_6::ImageBuf&, const OpenImageIO::v1_6::ImageBuf&, bool, OpenImageIO::v1_6::ROI, int)’:
/home/scott/Repositories/oiio/src/libOpenImageIO/imagebufalgo.cpp:325:21: error: variable ‘K’ set but not used [-Werror=unused-but-set-variable]
     const ImageBuf *K = &kernel;
                     ^
cc1plus: all warnings being treated as errors
make[2]: *** [src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/imagebufalgo.cpp.o] Error 1
make[1]: *** [src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/all] Error 2
make: *** [all] Error 2
```

I would delete the line, but I'm having troubles figuring out if the block of code that is related to K is needed or not. Is there some C++ magic that I'm not seeing, or is the temp kernel variable being used for anything?

For reference, here's the block of code:
```
bool
ImageBufAlgo::convolve (ImageBuf &dst, const ImageBuf &src,
                        const ImageBuf &kernel, bool normalize,
                        ROI roi, int nthreads)
{
    if (! IBAprep (roi, &dst, &src, IBAprep_REQUIRE_SAME_NCHANNELS))
        return false;
    bool ok;
    const ImageBuf *K = &kernel;
    ImageBuf Ktmp;
    if (kernel.spec().format != TypeDesc::FLOAT) {
        Ktmp.copy (kernel, TypeDesc::FLOAT);
        K = &Ktmp;
    }
    OIIO_DISPATCH_COMMON_TYPES2 (ok, "convolve", convolve_,
                          dst.spec().format, src.spec().format,
                          dst, src, kernel, normalize, roi, nthreads);
    return ok;
}
```

--- a/src/libOpenImageIO/imagebufalgo.cpp
+++ b/src/libOpenImageIO/imagebufalgo.cpp
@@ -330,7 +330,7 @@ ImageBufAlgo::convolve (ImageBuf &dst, const ImageBuf &src,
     }
     OIIO_DISPATCH_COMMON_TYPES2 (ok, "convolve", convolve_,
                           dst.spec().format, src.spec().format,
-                          dst, src, kernel, normalize, roi, nthreads);
+                          dst, src, *K, normalize, roi, nthreads);
     return ok;
 }
 

